lib/generators/boring/pronto/base_generator.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'boring_generators/generator_helper'
+
+module Boring
+  module Pronto
+    class BaseGenerator < Rails::Generators::Base
+      desc "Adds pronto gem with various extensions"
+
+      class_option :extensions_to_skip,
+       type: :array,
+        aliases: "-se",
+        desc: "Skip one or list of extensions. Example usage `--extensions_to_skip=rubocop flay`",
+        enum: %w[brakeman flay reek rubocop],
+        default: []
+
+      include BoringGenerators::GeneratorHelper
+
+      def add_pronto_gem
+        say "Adding pronto gem", :green
+
+        Bundler.with_unbundled_env do
+          check_and_install_gem "pronto"
```

Comment:
it doesn't seem this run install from code. does it?

Reply:
We have created a new helper method that checks the Gemfile and return early if it finds the gem. Else it will just run the gem command provided by the rails generator.

```
    def check_and_install_gem(*args)
      gem_name, = args

      gem_file_content_array = File.readlines("Gemfile")

      gem_exists = gem_file_content_array.any? { |line| line.include?(gem_name) }

      if gem_exists
        say "#{gem_name} is already in the Gemfile, skipping it ...", :yellow
      else
        gem *args
      end
    end
```